test/jest_integration/jwt_multiauth.test.ts

```diff
@@ -0,0 +1,178 @@
+      }
+    );
+
+    console.log('Starting OpenSearchDashboards server..');
+    await root.setup();
+    await root.start();
+
+    const getConfigResponse = await wreck.get(
+      'https://localhost:9200/_plugins/_security/api/securityconfig',
+      {
+        rejectUnauthorized: false,
+        headers: {
+          authorization: ADMIN_CREDENTIALS,
+        },
+      }
+    );
+    const responseBody = (getConfigResponse.payload as Buffer).toString();
+    config = JSON.parse(responseBody).config;
+    const jwtConfig = {
+      http_enabled: true,
+      transport_enabled: true,
+      order: 0,
+      http_authenticator: {
+        challenge: false,
+        type: 'jwt',
+        config: {
+          signing_key: 'OTkwMTFkZjZlZjQwZTRhMmNkOWNkNmNjYjJkNjQ5ZTAK',
+          jwt_header: 'Authorization',
+          jwt_url_parameter: 'token',
+          jwt_clock_skew_tolerance_seconds: 30,
+          roles_key: 'roles',
+          subject_key: 'sub',
+        },
+      },
+      authentication_backend: {
+        type: 'noop',
+        config: {},
+      },
+    };
+    try {
+      config.dynamic!.authc!.jwt_auth_domain = jwtConfig;
+      config.dynamic!.authc!.basic_internal_auth_domain.http_authenticator.challenge = true;
+      config.dynamic!.http!.anonymous_auth_enabled = false;
+      await wreck.put('https://localhost:9200/_plugins/_security/api/securityconfig/config', {
+        payload: config,
+        rejectUnauthorized: false,
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: ADMIN_CREDENTIALS,
+        },
+      });
+    } catch (error) {
+      console.log('Got an error while updating security config!', error.stack);
+      fail(error);
+    }
+  });
+
+  afterAll(async () => {
+    console.log('Remove the Security Config');
+    await wreck.patch('https://localhost:9200/_plugins/_security/api/securityconfig', {
+      payload: [
+        {
+          op: 'remove',
+          path: '/config/dynamic/authc/jwt_auth_domain',
+        },
+      ],
+      rejectUnauthorized: false,
+      headers: {
+        'Content-Type': 'application/json',
+        authorization: ADMIN_CREDENTIALS,
+      },
+    });
+    // shutdown OpenSearchDashboards server
+    await root.shutdown();
+  });
+
+  it('Verify JWT access to dashboards', async () => {
+    console.log('Wreck access home page');
+    const adminJWT = await new SignJWT({
+      roles: [JWT_ADMIN_ROLE],
+      sub: ADMIN_USER,
+    })
+      .setProtectedHeader({ alg: 'HS256' })
+      .sign(new TextEncoder().encode('99011df6ef40e4a2cd9cd6ccb2d649e0'));
+    await wreck.get(`http://localhost:5601/app/home?token=${adminJWT}#`, {
+      rejectUnauthorized: true,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+  });
+
+  it('Verify access to login page without JWT', async () => {
+    console.log('Wreck access login page without JWT');
+    await wreck.get('http://localhost:5601/app/login', {
```

Comment:
wdyt about having an assertion similar to [this](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/test/jest_integration/basic_auth.test.ts#L243-L248) for when token is not present as a url param?

```
const response = wreck.get('http://localhost:5601/app/home#/', ...)

expect(response.status).toEqual(302);
expect(response.header.location).toEqual('/app/login?nextUrl=%2Fapp%2Fhome');
```

Reply:
Added this.